pints/_nested/_nestedEllipsoid.py

```diff
@@ -0,0 +1,313 @@
+        
+        ## store weights
+        w = np.zeros(self._activePoints + self._iterations)
+        
+        ## store X values (defined in [1])
+        X = np.zeros(self._iterations+1)
+        X[0] = 1
+        
+        ## log marginal likelihood holder
+        vLogZ = np.zeros(self._iterations+1)
+        
+        print 'Starting rejection sampling...'
+        ## run iter
+        for i in range(0,self._iterations):
+            aRunningLogLikelihood = np.min(mActive[:,d])
+            aMinIndex = np.argmin(mActive[:,d])
+            X[i+1] = np.exp(-(i+1.0)/self._activePoints)
+            w[i] = X[i] - X[i+1]
+            vLogZ[i] = aRunningLogLikelihood
+            mInactive[i,:] = mActive[aMinIndex,:]
+    
+            if (i+1) % self._rejectionSamples == 0:
+                print 'Starting ellipsoidal sampling (finished rejection)...'
+                A, centroid = mvee(mActive[:,:d])
+            if i > self._rejectionSamples:
+                if (i + 1 -self._rejectionSamples) % self._ellipsoidUpdateGap == 0:
+                    print str(i+1 - self._rejectionSamples) + ' ellipsoidal samples completed (updating ellipsoid)...'
+                    A, centroid = mvee(mActive[:,:d])
+                    
+    
+            if i < self._rejectionSamples:
+                mActive[aMinIndex,:] = rejectSamplePrior(aRunningLogLikelihood,self._log_likelihood,self._prior)
+            else:
+                mActive[aMinIndex,:] = rejectEllipsoidSample_faster(aRunningLogLikelihood,self._log_likelihood,mActive[:,:d],self._enlargementFactor,A,centroid)
+
+        vLogZ[self._iterations] = logsumexp(mActive[:,d])
+        w[self._iterations:] = float(X[self._iterations]) / float(self._activePoints)
+        mSamplesAll = np.vstack((mInactive,mActive))
+        logZ = logsumexp(vLogZ, b = w[0:(self._iterations+1)])
+        vLogP = mSamplesAll[:,d] - logZ + np.log(w)
+        vP = np.exp(mSamplesAll[:,d] - logZ) * w
+        mTheta = mSamplesAll[:,:-1]
+        vIndex = np.random.choice(range(0,(self._iterations + self._activePoints)), self._numPosteriorSamples, p=vP)
+        mPosteriorSamples = mTheta[vIndex,:]
+        return [mPosteriorSamples, logZ]
+
+    def set_activePoints_rate(self, activePoints):
+        """
+        Sets the number of active points for the next run.
+        """
+        activePoints = float(activePoints)
+        if activePoints <= 5:
+            raise ValueError('Number of active points must be greater than 5.')
+        self._activePoints = activePoints
+
+    def set_iterations(self, iterations):
+        """
+        Sets the total number of iterations to be performed in the next run
+        (including burn-in and non-adaptive iterations).
+        """
+        iterations = int(iterations)
+        if iterations < 0:
+            raise ValueError('Number of iterations cannot be negative.')
+        self._iterations = iterations
+
+    def set_numPosteriorSamples(self,numPosteriorSamples):
+        
```

Comment:
Please add docstrings! These will then be used to generate the documentation on https://pints.readthedocs.io